src/StructureFactors/BasisReduction.jl

```diff
@@ -1,13 +1,13 @@
 function phase_averaged_elements(data, k::Vec3, sf::StructureFactor{N}, ffdata::Vector{FormFactor{FFType}}, ::Val{NCorr}, ::Val{NAtoms}) where {FFType, N, NCorr, NAtoms} 
-    elems = zero(SVector{NCorr, ComplexF64})
+    elems = zero(MVector{NCorr,ComplexF64})
     knorm = norm(k)
     ffs = ntuple(i -> compute_form(knorm, ffdata[i]), NAtoms)
     rs = ntuple(i -> sf.crystal.latvecs * sf.crystal.positions[i], NAtoms)
 
     for j in 1:NAtoms, i in 1:NAtoms
         phase = exp(im*(k ⋅ (rs[j] - rs[i])))
-        elems += phase * ffs[i] * ffs[j] * view(data, :, i, j)  # This view allocates
+        elems .+= phase .* ffs[i] .* ffs[j] .* view(data,:, i, j)
```

Comment:
Nope, it changes the meaning because it tries to do `view.(data,:,i,j)` which doesn't make sense (no method `view(::ComplexF64,::Colon,...)`)

So it can either be the `@.`, or use `view(...)` instead of `@view`, but not both.

Reply:
Nope, it changes the meaning because it tries to do `view.(data,:,i,j)` which doesn't make sense (no method `view(::ComplexF64,::Colon,...)`)

So it can either be the `@.`, or use `view(...)` instead of `@view`, but not both.

Reply:
I see. Let's go with the explicit `.` as you have written them.